readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -232,6 +229,34 @@ def build(self):
         )
         return cmd_ret.successful
 
+    def venv_sphinx_version(self):
+        """
+        Execute Python from the venv to query the ``sphinx`` version.
+
+        Example,
+
+            >>> import sphinx
+            >>> print(sphinx.__version__)
+            '2.0.0'
+
+        In this case, the output will be ``2.0.0``.
+        """
+
+        command = [
+            self.python_env.venv_bin(filename='python'),
+            '-c'
+            '"import sphinx; print(sphinx.__version__)"',
```

Comment:
when `DOCKER_ENABLED` is false, this will break. Because of the double quotes

```python
>>> proc = subprocess.Popen(['python3', '-c', 'import sphinx; print(sphinx.__version__)'], stdou
t=subprocess.PIPE)
>>> proc.communicate()
(b'1.8.5\n', None)
>>> proc = subprocess.Popen(['python3', '-c', '"import sphinx; print(sphinx.__version__)"'], std
out=subprocess.PIPE)
>>> proc.communicate()
(b'', None)

```

Reply:
Thanks for raising this.

I made it work by using `shell=True` and flattening the command when `shell=True` using the helper that we had there.

I like how the ended now. On `BuildCommand` it uses the `shell=True` attribute and flatten the command to be compatible. On `DockerBuildCommand` it uses `escape_command=False` to not escape the command.

It works in both builder backends.